index.js

```diff
@@ -1,137 +1,130 @@
 const core = require('@actions/core')
 const github = require('@actions/github')
 
-async function base2HeadUpdate () {
-  try {
-    // Get the JSON webhook payload for the event that triggered the workflow
-    const payloadStr = JSON.stringify(github.context, undefined, 2)
-    console.log(`The event payload: ${payloadStr}`)
+class Base2HeadUpdate {
+  shouldRun (prObj) {
+    const pullNumber = prObj.number
+    const labels = prObj.labels
+    let hasSkipLabel = false
+    let hasActLabel = false
+    for (let li = 0; li < labels.length; li++) {
+      const labelFound = labels[li].name.toLowerCase()
+      if (this.blockedLabels.includes(labelFound)) {
+        hasSkipLabel = true
+        console.log(`Pull #${pullNumber} has skip label: ${labelFound}!`)
+        break
+      } else if (!!this.actLabel && labelFound === this.actLabel.toLowerCase()) {
+        hasActLabel = true
+        console.log(
+          `Pull ${pullNumber} has act label: ${labelFound}, continue looking for skip labels`
+        )
+      }
+    }
+    const autoMergeEnabled = (
+      ('auto_merge' in prObj) && (!!prObj.auto_merge) && (!!prObj.auto_merge.enabled_by)
+    )
+    if (autoMergeEnabled) {
+      console.log(`Automerge enabled by: ${prObj.auto_merge.enabled_by.login}`)
+    }
 
-    const repoToken = core.getInput('repo-token')
-    const workingLabel = core.getInput(
-      'act-label'
+    return !hasSkipLabel && (
+      hasActLabel || (
+        autoMergeEnabled
+      )
```

Comment:
Optional(or future): Maybe we can actively ignore "draft", wip, and do-not-merge labels from being merged or updated? It can be a parameter to define these labels too.

Reply:
Ok i see it's already there. NVM

Reply:
draft, yes. Wip and do not merge labels is already implemented.